nixos/doc/manual/development/writing-nixos-tests.section.md

```diff
@@ -313,3 +315,52 @@ repository):
       # fmt: on
     '';
 ```
+
+## Failing tests early
+
+To fail tests early when certain invariables are no longer met (instead of waiting for the build to time out), the decorator `polling_condition` is provided. For example, if we are testing a program `foo` that should not quit after being started, we might write the following:
+
+```py
+@polling_condition
+def foo_running():
+    machine.succeed("pgrep -x foo")
+
+
+machine.succeed("foo --start")
+machine.wait_until_succeeds("pgrep -x foo")
+
+with foo_running:
+    ...  # Put `foo` through its paces
+```
+
+
+`polling_condition` takes the folloing (optional) arguments:
+
+`every`
+
+:
+    specifies how often (in seconds) the condition should be polled:
+
+    ```py
+    @polling_condition(every=10)
```

Comment:
can we rename this to `every_seconds` or `seconds_interval` to have the unit there, too?

Reply:
I think it's pretty standard to have the unit be seconds for time if it's not specified...

Reply:
I've seen milliseconds and microseconds be the unit for sleep or other time intervals.
Explicit is good. Code that's optimized for reading/understanding is much more pleasant to debug.

Reply:
That's true I guess. I'll rename to `seconds_interval`